src/test/java/api/loans/ScheduledDueDateNoticesProcessingTests.java

```diff
@@ -254,6 +255,82 @@ public void processingTakesNoticesInThePastLimitedAndOrdered()
     assertThat(unprocessedScheduledNotices, hasItems(noticesInTheFuture.toArray(new JsonObject[0])));
   }
 
+  @Test
+  public void testNumberOfProcessedNoticesWithSchedulerNoticesLimitConfiguration()
+    throws MalformedURLException,
+    InterruptedException,
+    TimeoutException,
+    ExecutionException {
+
+    scheduledNoticesClient.deleteAll();
+
+    int noticesLimitConfig = 200;
+    int numberOfNotices = 300;
+    int expectedNumberOfUnprocessedNotices = numberOfNotices - noticesLimitConfig;
+
+    // create a new configuration
+    configClient.create(ConfigurationExample.schedulerNoticesLimitConfiguration(Integer.toString(noticesLimitConfig)));
+
+    createNotices(numberOfNotices);
+    scheduledNoticeProcessingClient.runNoticesProcessing();
+    List<JsonObject> unprocessedScheduledNotices = scheduledNoticesClient.getAll();
+
+    assertThat(unprocessedScheduledNotices, hasSize(expectedNumberOfUnprocessedNotices));
+  }
+
+  @Test
+  public void testNumberOfProcessedNotificationsWithIncorrectConfiguration()
+    throws MalformedURLException,
+    InterruptedException,
+    TimeoutException,
+    ExecutionException {
+
+    scheduledNoticesClient.deleteAll();
+
+    int numberOfNotices = 259;
+    int expectedNumberOfUnprocessedNotices = numberOfNotices - SCHEDULED_NOTICES_PROCESSING_LIMIT;
+
+    // create a incorrect configuration
+    configClient.create(ConfigurationExample.schedulerNoticesLimitConfiguration("IncorrectVal"));
```

Comment:
Does this line setup the mock configuration once and the subsequent tests, such as  testDefaultNumberOfProcessedNotices, will be able to make use of this setup? Or does each test need to set up the mock configuration separately?

Reply:
In the test environment (FakeOkapi.class), there is one default mock implementation of the mod-configuration. It doesn't contain any data.
For each test, we need to configure the mock configuration separately.

Reply:
Does it mean that the test _testDefaultNumberOfProcessedNotices_ needs to have this line: configClient.create(ConfigurationExample.schedulerNoticesLimitConfiguration("_SomeCorrectValue_")); ?

Reply:
I run the tests in the `ScheduledDueDateNoticesProcessingTests` class and made sure that every time the mock mod-config does not have any data. And to determine the state it needs to be configured `configClient.create(ConfigurationExample.schedulerNoticesLimitConfiguration("SomeCorrectValue"));`

Reply:
I check the default `DEFAULT_SCHEDULED_NOTICES_PROCESSING_LIMIT` value with the `testDefaultNumberOfProcessedNotices` test in the case when we do not have a configuration in the mod-config.

Reply:
To do this, we do not need to configure the module with `configClient.create(ConfigurationExample.schedulerNoticesLimitConfiguration("SomeCorrectValue"))`

Reply:
ok, thanks.